crates/ruff_cli/src/diagnostics.rs

```diff
@@ -177,7 +177,23 @@ impl AddAssign for FixMap {
     }
 }
 
+fn get_override_source_type(
+    path: Option<&Path>,
+    extension: &FxHashMap<String, Language>,
+) -> Option<PySourceType> {
+    let Some(ext) = path.and_then(|p| p.extension()).and_then(|p| p.to_str()) else {
+        return None;
+    };
+    match extension.get(ext) {
+        Some(Language::Python) => Some(PySourceType::Python),
+        Some(Language::Ipynb) => Some(PySourceType::Ipynb),
+        Some(Language::Pyi) => Some(PySourceType::Stub),
+        None => None,
+    }
```

Comment:
nit: we could probably implement `impl From<Language> for PySourceType` and use that here like `extension.get(ext).map(PySourceType::from)`.

Reply:
done